sklearn/preprocessing/_encoders.py

```diff
@@ -696,7 +730,12 @@ def transform(self, X):
         X_out : sparse matrix or a 2-d array
             Transformed input.
         """
-        X_int, _ = self._transform(X)
+        X_int, X_mask = self._transform(X, handle_unknown=self.handle_unknown)
+
+        # create separate category for unknown values
+        if self.handle_unknown == 'use_encoded_value':
+            for i in range(len(self.categories_)):
+                X_int[~X_mask[:, i], i] = self.unknown_value
```

Comment:
do we need to loop over features here? Can't we just do `X_int[~X_mask] =self.unknown_value`?